scipy/optimize/nonlin.py

```diff
@@ -1227,7 +1227,7 @@ def __init__(self, alpha=None):
 
     def setup(self, x, F, func):
         GenericBroyden.setup(self, x, F, func)
-        self.d = np.ones((self.shape[0],), dtype=self.dtype) / self.alpha
+        self.d = np.full((self.shape[0],), 1 / self.alpha, dtype=self.dtype) 
```

Comment:
Might be different if `self.dtype` can be integral.

Reply:
In this case I suspect that it's a bug and they meant for `self.d` to have the same dtype, even though dividing by `self.alpha` might change it. So I'm okay with it as is.